### PR TITLE
Exit with code 2 on blocking errors

### LINKED ISSUE #2754: Use a different error code on early exit
When running mypy from a script, it's sometimes nice to know when mypy successfully completed a full typechecking run vs exited early due to blocking errors.  Perhaps mypy could exit with exit code `2` when exiting due to blocking errors?

### PR DESCRIPTION
**Pull request by msullivan**
Exit with code 2 on blocking errors, while continuing to exit with 1
when there are nonblocking errors.  Add test infrastructure.

Fixes #2754.

### CODE REVIEW

**Comment by msullivan** [comment]
(This depends on #4396 because they would otherwise conflict)

File: mypy/test/testcmdline.py [hunk 1]
```diff
@@ -78,6 +79,7 @@ def test_python_cmdline(testcase: DataDrivenTestCase) -> None:
                                            path))
     else:
         out = normalize_error_messages(out)
+        out.append('== Return code: {}'.format(result))
```
Comment (gvanrossum) [inline mypy/test/testcmdline.py]: Hm, this added a lot of noise to a lot of tests. Perhaps only print the return code when it's not equal to `not len(out)`?

**Comment by msullivan** [comment]
Those cases don't add blockers to the errors object, but they raise CompileError directly, which causes us to exit with 2. One of the tests I added checks for this.

**Comment by gvanrossum** [comment]
Ah, I see. But I'd still like to see fewer tests with different output.

**Comment by gvanrossum** [comment]
Whee!

**Pull request merged**